airbyte-integrations/bases/debezium/src/main/java/io/airbyte/integrations/debezium/internals/AirbyteSchemaHistoryStorage.java

```diff
@@ -127,8 +218,9 @@ public static AirbyteSchemaHistoryStorage initializeDBHistory(final Optional<Jso
     }
     final Path dbHistoryFilePath = dbHistoryWorkingDir.resolve("dbhistory.dat");
 
-    final AirbyteSchemaHistoryStorage schemaHistoryManager = new AirbyteSchemaHistoryStorage(dbHistoryFilePath);
-    schemaHistoryManager.persist(schemaHistory);
+    final AirbyteSchemaHistoryStorage schemaHistoryManager =
+        new AirbyteSchemaHistoryStorage(dbHistoryFilePath, schemaHistoryInfo.compressionEnabled());
+    schemaHistoryManager.persist(schemaHistoryInfo.schemaHistory(), schemaHistoryInfo.isSchemaHistoryCompressed());
```

Comment:
does this mean a stream will not switch over from compressed : false to compressed : true when the global flag is turned on?

Reply:
So the `schemaHistoryInfo.isSchemaHistoryCompressed ()` defines whether the saved state that the sync received is compressed or not. 
The `schemaHistoryInfo.compressionEnabled()` defines whether to output a compressed blob in the state or not.